ImageProcessor/src/main/java/de/telran/service/ImageFileService.java

```diff
@@ -0,0 +1,30 @@
+package de.telran.service;
+
+import de.telran.entity.ActionableImage;
+
+import javax.imageio.ImageIO;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class ImageFileService {
+
+    private ConfigService configService;
+
+    public ImageFileService(ConfigService configService){
+        this.configService = configService;
+    }
+
+    public void saveImageAsFile(ActionableImage image) {
+        try {
+            ImageIO.write(image.getImage(), "jpg",new File(configService.getPathToSavedImages()
+                    + image.hashCode()+ ".jpg"));
```

Comment:
You already have the original URL of the image, so, you may employ it here for generating a file name out of it instead of the hash.